core/services/relay/evm/chain_writer.go

```diff
@@ -71,6 +79,11 @@ type chainWriter struct {
 	encoder commontypes.Encoder
 }
 
+type CWGasEstimator struct {
+	estimator   gas.EvmFeeEstimator
+	maxGasPrice *big.Int
+}
```

Comment:
Why are these packaged together? Would it be simpler to just have two field on the chainWriter?

Reply:
Good point, I'll remove the `CWGasEstimator` struct